oshmem/include/pshmemx.h

```diff
@@ -16,110 +16,194 @@
 extern "C" {
 #endif
 
+
+/*
+ * Legacy API
+ * old init/destruct functions - not in the open shmem spec but still supported
+ */
+OSHMEM_DECLSPEC  void pstart_pes(int npes);
+
+OSHMEM_DECLSPEC  int p_num_pes(void);
+OSHMEM_DECLSPEC  int p_my_pe(void);
+
+OSHMEM_DECLSPEC  void* pshmalloc(size_t size);
+OSHMEM_DECLSPEC  void* pshmemalign(size_t align, size_t size);
+OSHMEM_DECLSPEC  void* pshrealloc(void *ptr, size_t size);
+OSHMEM_DECLSPEC  void pshfree(void* ptr);
+
+OSHMEM_DECLSPEC  void pshmem_char_put(char *target, const char *source, size_t len, int pe);
+OSHMEM_DECLSPEC  void pshmem_char_get(char *target, const char *source, size_t len, int pe);
```

Comment:
@igor-ivanov Ok, no prob.  Just wanted to make you aware that this kind of capability is available.  Seems like this PR should be merged in, and if you guys later want to add deprecation pragmas, that can be done in a future PR.  Sound good?